src/grasp_scorer.cpp

```diff
@@ -110,6 +110,89 @@ Eigen::Vector3d GraspScorer::scoreGraspTranslation(const Eigen::Affine3d& grasp_
   return scores;
 }
 
+Eigen::Vector3d GraspScorer::scoreGraspOverhang(const Eigen::Affine3d& grasp_pose, const GraspDataPtr& grasp_data,
+                                                const Eigen::Affine3d& object_pose, const Eigen::Vector3d& object_size)
+{
+  Eigen::Vector3d scores(0, 0, 0);
+
+  Eigen::Affine3d gripper_corner_tr = Eigen::Affine3d::Identity();
+  Eigen::Affine3d gripper_corner_tl = Eigen::Affine3d::Identity();
+  Eigen::Affine3d gripper_corner_br = Eigen::Affine3d::Identity();
+  Eigen::Affine3d gripper_corner_bl = Eigen::Affine3d::Identity();
+
+  Eigen::Affine3d box_corner_tr = Eigen::Affine3d::Identity();
+  Eigen::Affine3d box_corner_tl = Eigen::Affine3d::Identity();
+  Eigen::Affine3d box_corner_br = Eigen::Affine3d::Identity();
+  Eigen::Affine3d box_corner_bl = Eigen::Affine3d::Identity();
+
+  gripper_corner_tr.translation() =
```

Comment:
This would reduce unnecessary 3d computations
```suggestion
Eigen::Affine3d gripper_pose = object_pose.inverse() * grasp_pose;
Eigen::Affine2d gripper_pose_2d =
          Eigen::Translation2d(gripper_pose.translation().topRows<2>()) *
                               gripper_pose.linear().topLeftCorner<2,2>();
Eigen::Vector2d gripper_corner_tr(grasp_data->active_suction_range_x_ / 2,
                                   grasp_data->active_suction_range_y_ / 2);
gripper_corner_tr = gripper_pose_2d * gripper_corner_tr;
    
```